raycast/BaseGame.py

```diff
@@ -0,0 +1,145 @@
+import collections
+import time
+from abc import ABCMeta
+
+import katagames_sdk.engine as kataen
+
+
+pygame = kataen.import_pygame()
+EventReceiver = kataen.EventReceiver
+EngineEvTypes = kataen.EngineEvTypes
+
+
+class BaseGame(metaclass=ABCMeta):
+    """
+    Base class for games, this is staged for the katasdk0.0.7 release
+    """
+
+    FPS_TRACKING_DEFAULT_SS = 13
+
+    def __init__(self, track_fps=True):
+        if track_fps:
+            self._fps_n_frames = self.FPS_TRACKING_DEFAULT_SS
+        else:
+            self._fps_n_frames = 0
+
+        self._fps_tracker_logic = collections.deque()
+        self._fps_tracker_rendering = collections.deque()
```

Comment:
So here's a question I was wondering about - is it ever possible for rendering and logic updates to _not_ just simply alternate? I structured the FPS tracking this way because it seemed like they might have been asynchronous from one another but I'm wondering whether that's actually true? If they always just alternate, then this FPS tracking can probably just be done with a single deque, and the `logical=True` arg in `get_fps` can be removed

Reply:
they always alternate as long as custom "EventReceiver" objects do not mess with events taken directly from the `EngineEvTypes` category.
The LOGICUPDATE event is sent first, followed by a PAINT event. I remove the `logical=True`